docs/adr/0003-git-workflow.md

```diff
@@ -0,0 +1,73 @@
+
+
+## Decision Drivers
+
+* Increase the velocity of progress
+* Pave the way for faster mainnet release cycles
+* Have certain recent checkpoints in the project's codebase where new features / functionalities are known to be stable.
+
+
+## Considered Options
+
+### Original proposal
+After discussing this with @hackaugusto, @LefterisJP and @konradkonrad, we realized that we need to come up with a better way to separate our workflow for bug fixing from the ones for introducing new features and refactoring of the code base.
+
+So, we seem to agree on using parts of the git-workflow where:
+1. A `develop` branch is introduced.
+2. The `master` branch will be our "production-ready" branch.
+3. When a bug is reported, the assignee of the issue will branch of the latest master, provide a fix which will be merged to this master branch.
+4. The `develop` branch will contain all the work we do for new features and code refactorings.
+5.  Once any code changes are merged to master, `develop` should be rebased on top of the latest master to ensure that our fixes are also residing in the develop branch in case any team member is working on the same code base.
+
+The Pros of this new workflow:
+1. Resolving bugs, merging them to master and instantly creating a release which contains the new bug fix(es)
+2. Removing the worry about a broken `master` branch because the `develop` branch is going to contain all the potential breaking changes whose issues will be resolved separately.
+
+The Cons of this approach:
+1. Rebasing `develop` should some how be automated as soon as any changes are merged into master, or otherwise if both branches diverge alot from each other, this could be a pain to resolve all the conflicts and get the develop branch back in shape.
+2. It's a new workflow so team members have to get used to this and map what they're working on to the destination branch on which their implementation will land.
+
+### "Successful branching model"
+
+It was suggested to follow [The successful git branching model](https://nvie.com/posts/a-successful-git-branching-model/) which is quite similar to the original proposal.
+
+## Decision Outcome
+
+It was decided to follow the "Successful git branching model" option which is conveyed by the following:
+
+![git branching mode](https://user-images.githubusercontent.com/44281/56030116-5cdd8680-5d1c-11e9-8de8-d3fb61ad7d8e.png)
+
+Where:
+
+- `master` is maintained to be production-ready branch.
+- `develop` is the target branch for new features.
+- New feature branches are merged into develop.
+- Bug fixes are merged into both `develop` and `master`.
+- Nightly builds are created on latest `develop`.
```

Comment:
Additionally, every bugfix is a minor release.

Reply:
This refers to nightly builds. But i think we already have an internal agreement about how to version releases right?

Reply:
We have an agreement on the version scheme, semantic versioning, the frequency os releases are being discussed AFAIK

Reply:
Added the point you asked for.